ci-operator/step-registry/upi/deprovision/aws/stage1/upi-deprovision-aws-stage1-commands.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+set +e
+
+function queue() {
+    local TARGET="${1}"
+    shift
+    local LIVE
+    LIVE="$(jobs | wc -l)"
+    while [[ "${LIVE}" -ge 45 ]]; do
+    sleep 1
+    LIVE="$(jobs | wc -l)"
+    done
+    echo "${@}"
+    if [[ -n "${FILTER:-}" ]]; then
+    "${@}" | "${FILTER}" >"${TARGET}" &
+    else
+    "${@}" >"${TARGET}" &
+    fi
+}
+
+
+export PATH=$PATH:/tmp/shared
+AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred
+export AWS_SHARED_CREDENTIALS_FILE
+
+echo "Deprovisioning cluster ..."
+export PATH="${HOME}/.local/bin:${PATH}"
+
+AWS_REGION=$(cat ${SHARED_DIR}/AWS_REGION)
+AWS_DEFAULT_REGION=${AWS_REGION}  # CLI prefers the former
+export AWS_REGION
+export AWS_DEFAULT_REGION
+CLUSTER_NAME=$(cat ${SHARED_DIR}/CLUSTER_NAME)
+
+aws --region $AWS_REGION cloudformation describe-stack-resources --stack-name "${CLUSTER_NAME}-control-plane" \
+    --query 'StackResources[?ResourceType==`AWS::EC2::Instance`].PhysicalResourceId' --output text | sed 's,\t,\n,g' > /tmp/node-provider-IDs
+for INDEX in 0 1 2
+do
+    aws --region $AWS_REGION cloudformation describe-stack-resources --stack-name "${CLUSTER_NAME}-compute-${INDEX}" \
+    --query 'StackResources[].PhysicalResourceId' --output text | cut -d, -f1 >> /tmp/node-provider-IDs
```

Comment:
The current approach for `gather-aws-console` is to populate `${SHARED_DIR}/aws-instance-ids.txt` as you create the instances, e.g. [here][1].  That way, `gather-aws-console` can focus on going from instance IDs to logs, and doesn't have to understand the various ways previous steps may have provisioned instances (Terraform, CloudFormation, etc.).

[1]: https://github.com/openshift/release/blob/7717497fc639610d4b537752d50408236f15a610/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L19